app/src/main/java/intellistart/interviewplanning/controllers/rest/InterviewerController.kt

```diff
@@ -32,34 +30,40 @@ class InterviewerController(
     fun createInterviewerSlot(
         @RequestBody slotDto: SlotDto,
         authentication: Authentication
-    ): ResponseEntity<SlotDto> {
-        val interviewerSlot = getInterviewerSlotFromDto(slotDto)
+    ): Mono<SlotDto> = Mono.defer {
         val jwtUserDetails = authentication.principal as JwtUserDetails
+        val interviewerSlot = getInterviewerSlotFromDto(slotDto)
         slotValidator.validateCreating(interviewerSlot, jwtUserDetails.email)
-        val createdInterviewerSlot = slotService.create(interviewerSlot, jwtUserDetails.email)
-        return ResponseEntity(createdInterviewerSlot.toDto(), HttpStatus.OK)
+            .then(slotService.create(interviewerSlot, jwtUserDetails.email))
+            .map {
+                it.toDto()
+            }
     }
 
     @PostMapping("/interviewer/slot/update/{slotId}")
     fun updateInterviewerSlot(
         @RequestBody slotDto: SlotDto,
         @PathVariable("slotId") slotId: String,
         authentication: Authentication
-    ): ResponseEntity<SlotDto> {
-        val interviewerSlot = getInterviewerSlotFromDto(slotDto).copy(id = ObjectId(slotId))
+    ): Mono<SlotDto> = Mono.defer {
         val jwtUserDetails = authentication.principal as JwtUserDetails
+        val interviewerSlot = getInterviewerSlotFromDto(slotDto).copy(id = ObjectId(slotId))
         slotValidator.validateUpdating(interviewerSlot, jwtUserDetails.email)
-        val updatedInterviewerSlot = slotService.update(interviewerSlot, jwtUserDetails.email)
-        return ResponseEntity(updatedInterviewerSlot.toDto(), HttpStatus.OK)
+            .then(slotService.update(interviewerSlot, jwtUserDetails.email))
+            .map {
+                it.toDto()
+            }
     }
 
     @GetMapping("/interviewer/slots")
-    fun getInterviewerSlotsForNextWeek(
+    fun getAllInterviewerSlots(
         authentication: Authentication
-    ): ResponseEntity<SlotsDto> {
+    ): Flux<SlotDto> {
         val jwtUserDetails = authentication.principal as JwtUserDetails
-        val slots = slotService.getAllSlotsByEmail(jwtUserDetails.email)
-        return ResponseEntity.ok(slots.toDtoList())
+
+        return slotService.getAllSlotsByEmail(jwtUserDetails.email)
+            .map { it.toDto() }
+            .defaultIfEmpty(SlotDto())
```

Comment:
```suggestion
            .switchIfEmpty { SlotDto().toMono() }
```
Don't build redundant object all the time